orchagent/pfcactionhandler.cpp

```diff
@@ -266,11 +283,30 @@ PfcWdAclHandler::~PfcWdAclHandler(void)
 {
     SWSS_LOG_ENTER();
 
-    auto found = m_aclTables.find(m_strIngressTable);
-    found->second.unbind(getPort());
+    sai_object_id_t table_oid = gAclOrch->getTableById(m_strIngressTable);
+    map<sai_object_id_t, AclTable> table_map = gAclOrch->getAclTables();
+    auto rule_iter = table_map[table_oid].rules.find(m_strRule);
+    uint64_t table_size = table_map[table_oid].rules.size();
 
-    found = m_aclTables.find(m_strEgressTable);
-    found->second.unbind(getPort());
+    vector<sai_object_id_t> port_set = rule_iter->second->getInPorts();
+    sai_object_id_t port = getPort();
+
+    if ((port_set.size() == 1) && (port_set[0] == port))
+    {
+        gAclOrch->removeAclRule(m_strIngressTable, m_strRule);
+        if (table_size == 1) 
+        {
+            auto found = m_aclTables.find(m_strIngressTable);
+            found->second.unbind(port);
```

Comment:
I am not sure if this is the right approach to unbind. we will need to unbind a port if its not used by any  rules in the table. @daall , @prsunny , can you confirm if we really need to bind the port to the table since now we are including it in the rule.

Reply:
Agree with @neethajohn, that this may cause issues to existing rules in that table.